src/tls_wrap.cc

```diff
@@ -875,20 +886,26 @@ int TLSWrap::SelectSNIContextCallback(SSL* s, int* ad, void* arg) {
   p->sni_context_.Reset(env->isolate(), ctx);
 
   SecureContext* sc = Unwrap<SecureContext>(ctx.As<Object>());
+  CHECK_NE(sc, nullptr);
   p->SetSNIContext(sc);
   return SSL_TLSEXT_ERR_OK;
 }
 #endif  // SSL_CTRL_SET_TLSEXT_SERVERNAME_CB
 
 
+void New(const FunctionCallbackInfo<Value>& args) {
+  args.This()->SetAlignedPointerInInternalField(0, nullptr);
+}
+
+
 void TLSWrap::Initialize(Local<Object> target,
                          Local<Value> unused,
                          Local<Context> context) {
   Environment* env = Environment::GetCurrent(context);
 
   env->SetMethod(target, "wrap", TLSWrap::Wrap);
 
-  Local<FunctionTemplate> t = FunctionTemplate::New(env->isolate());
+  Local<FunctionTemplate> t = env->NewFunctionTemplate(New);
```

Comment:
The `New()` function that's not a TLSWrap method looks a little incongruous.  You could use a lambda instead:
```c++
auto constructor = [](const FunctionCallbackInfo<Value>& args) { /* ... */ };
auto t = env->NewFunctionTemplate(constructor);
```